terraform/aws/aws.tf

```diff
@@ -0,0 +1,199 @@
+variable "region" { }
+variable "secret_key" { }
+variable "access_key" { }
+variable "cluster_id" { }
+variable "network_ipv4" {default = "10.0.0.0/16"}
+variable "network_subnet_ip4" {default = "10.0.0.0/16"}
+variable "control_count" {default = "3"}
+variable "worker_count" {default = "1"}
+variable "control_type" {default = "m1.small"}
+variable "source_ami" { }
+variable "worker_type" {default = "m1.small"}
+variable "public_key" { }
+
+
+provider "aws" {
+  access_key = "${var.access_key}"
+  secret_key = "${var.secret_key}"
+  region = "${var.region}"
+}
+
+resource "aws_vpc" "main" {
+     cidr_block = "${var.network_ipv4}"
+}
+
+resource "aws_subnet" "main" {
+    vpc_id = "${aws_vpc.main.id}"
+    cidr_block = "${var.network_subnet_ip4}"
+}
+
+resource "aws_instance" "mi-control-nodes" {
+    ami = "${var.source_ami}"
+    instance_type = "${var.control_type}"
+    count = "${var.control_count}"
+    vpc_security_group_ids = ["${aws_security_group.external.id}",
+                              "${aws_vpc.main.default_security_group_id}"]
+
+    key_name = "${aws_key_pair.deployer.key_name}"
+
+    associate_public_ip_address=true
+
+    subnet_id = "${aws_subnet.main.id}"
+
+    tags {
+      Name = "${var.cluster_id}_terraform"
+    }
+}
+
+resource "aws_instance" "mi-worker-nodes" {
+    ami = "${var.source_ami}"
+    instance_type = "${var.worker_type}"
+    count = "${var.worker_count}"
+
+    vpc_security_group_ids = ["${aws_security_group.external.id}",
+                              "${aws_vpc.main.default_security_group_id}"]
+
+
+    key_name = "${aws_key_pair.deployer.key_name}"
+
+    subnet_id = "${aws_subnet.main.id}"
+
+    tags {
+      Name = "${var.cluster_id}_terraform"
+    }
+}
+
+resource "aws_security_group" "allow_all_icmp" {
+  name = "allow_all_icmp_${var.cluster_id}"
+  description = "Allow all traffic within our subnet"
```

Comment:
this allows all external traffic but says "within our subnet". Just a little fix needed here. :)